app/mutations/product_naics_code/updateProductNaicsCodeMutation.ts

```diff
@@ -0,0 +1,60 @@
+import {graphql} from 'react-relay';
+import RelayModernEnvironment from 'relay-runtime/lib/store/RelayModernEnvironment';
+import {
+  updateProductNaicsCodeMutation as updateProductNaicsCodeMutationType,
+  updateProductNaicsCodeMutationVariables
+} from 'updateProductNaicsCodeMutation.graphql';
+import BaseMutation from 'mutations/BaseMutation';
+import {ConnectionHandler, RecordSourceSelectorProxy} from 'relay-runtime';
+import {AllowableProductsTableRow_productNaicsCode} from '__generated__/AllowableProductsTableRow_productNaicsCode.graphql';
+
+const mutation = graphql`
+  mutation updateProductNaicsCodeMutation(
+    $input: UpdateProductNaicsCodeInput!
+  ) {
+    updateProductNaicsCode(input: $input) {
+      productNaicsCode {
+        id
+        ...AllowableProductsTableRow_productNaicsCode
+      }
+    }
+  }
+`;
+
+// TODO: May want to surface the onCompleted errors to the user (ie not reject, resolve & report)
+const updateProductNaicsCodeMutation = async (
+  environment: RelayModernEnvironment,
+  variables: updateProductNaicsCodeMutationVariables,
+  naicsCodeId: string,
+  optimisticProductNaicsCode: AllowableProductsTableRow_productNaicsCode,
+  connectionKey: string
+) => {
+  const optimisticResponse = {
+    updateProductNaicsCode: {
+      productNaicsCode: {
+        id: variables.input.id,
+        ...variables.input.productNaicsCodePatch,
+        ...optimisticProductNaicsCode
+      }
+    }
+  };
+
+  const updater = (store: RecordSourceSelectorProxy) => {
+    const record = store.get(naicsCodeId);
+    const connection = ConnectionHandler.getConnection(record, connectionKey);
+    ConnectionHandler.deleteNode(connection, variables.input.id);
+  };
```

Comment:
We're assuming that we're doing a delete here, which isn't necessarily the case. maybe this file should be renamed to `deleteProductNaicsCodeMutation`, and we can hardcode the `productNaicsCodePatch` to set the `deletedAt` variable

Reply:
That's a really good point, it would also make the API a lot clearer for the caller